core/src/test/scala/org/apache/spark/serializer/KryoSerializerSuite.scala

```diff
@@ -242,6 +244,15 @@ class KryoSerializerSuite extends FunSuite with SharedSparkContext {
       ser.newInstance().deserialize[ClassLoaderTestingObject](bytes)
     }
   }
+
+  test("registration of HighlyCompressedMapStatus") {
+    val conf = new SparkConf(false)
+    conf.set("spark.kryo.registrationRequired", "true")
+    val hcmo = HighlyCompressedMapStatus(BlockManagerId("exec-1", "host", 1234), Array(0l,2l,5l))
+    val ser = new KryoSerializer(conf)
+    val serInstance = ser.newInstance()
+    serInstance.serialize(hcmo)
```

Comment:
Would this actually fail without registration? My understanding is that the serialized data would contain class names instead of a more efficient identifier, but would still work.